File: 2. python/1. basics/lecture 2/p9.py
class Solution:
    def __init__(self, n): #constructor
        for i in range(n):
            print(" " * (n - i - 1), end="")
            print("* " * (2 * i + 1))
        for i in range(n - 2, -1, -1):
            print(" " * (n - i - 1), end="")
            print("* " * (2 * i + 1))

''' pattern output:

    *
  * * *
* * * * *
  * * *
    *

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("1. output:")
    Solution(size)

File: 1. cpp/1. basics/lecture 2/p9.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                cout << "   ";
            }
            for (int j = 0; j < 2 * i + 1; j++) {
                cout << " * ";
            }
            cout << endl;
        }
        for (int i = n - 2; i >= 0; i--) {
            for (int j = 0; j < n - i - 1; j++) {
                cout << "   ";
            }
            for (int j = 0; j < 2 * i + 1; j++) {
                cout << " * ";
            }
            cout << endl;
        }
    }
};

/* pattern output:

    *
  * * *
* * * * *
  * * *
    *

*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"9. output:"<<endl;
  solution pattern(size);

}